docs/zkapps/tutorials/anonymous-message-board.mdx

```diff
@@ -210,35 +223,30 @@ The `init` method is similar to the `constructor` in Solidity. It’s a place fo
 }
 ```
 
-### Defining `publishMessage()`
+### Define `publishMessage()`
 
-The `publishMessage` method will allow an approved user to publish a message. Note the `@method` decorator mentioned earlier. It makes this method callable by users so that they can interact with the smart contract.
+The `publishMessage` method allows an approved user to publish a message. The `@method` decorator makes this method callable by users so that they can interact with the smart contract.
 
-For our example, we’ll pass in `message` and `signerPrivateKey` arguments to check that the user holds a private key associated with one of the three on-chain public keys before allowing them to update the message.
+For this example, pass in `message` and `signerPrivateKey` arguments to check that the user holds a private key associated with one of the three on-chain public keys before allowing them to update the message:
 
 ```ts
-  @method publishMessage(message: Field, signerPrivateKey: PrivateKey) {
-    // Compute signerPublicKey from signerPrivateKey argument
-    // Get approved public keys
-    // Assert that signerPublicKey is one of the approved public keys
-    // Update on-chain message state
-    // Computer new messageHistoryHash
-    // Update on-chain messageHistoryHash
-  }
+@method publishMessage(message: Field, signerPrivateKey: PrivateKey) {
```

Comment:
```suggestion
@method async publishMessage(message: Field, signerPrivateKey: PrivateKey) {
```

Reply:
This will be new async circuit syntax after the next `o1js` release https://github.com/o1-labs/o1js/pull/1477.